dagmc_stats/DagmcQuery.py

```diff
@@ -7,61 +7,87 @@
+        """This class provides the functionality for making queries about
+        various metrics for the meshset(s) of interest.
 
         inputs
         ------
-        dagmc_file : DagmcFile instance
-        meshset: the meshset on which query will be performed. The rootset will be used by default.
+            dagmc_file : DagmcFile instance
+            meshset: the meshset on which query will be performed.
+                The rootset will be used by default.
 
         outputs
         -------
-        none
+            none
         """
         self.dagmc_file = dagmc_file
         self.meshset_lst = []
         if meshset is None:
-            self.meshset_lst.append(self.dagmc_file.root_set)
+            self.meshset_lst =  list(self.dagmc_file.entityset_ranges['surfaces'])
         else:
             self.__get_entities(meshset)
+        self.tris = self.__get_tris()
+        self.verts = self.__get_verts()
         # initialize data frames
         self._vert_data = pd.DataFrame()
         self._tri_data = pd.DataFrame()
         self._surf_data = pd.DataFrame()
         self._vol_data = pd.DataFrame()
+        self._tri_vert_data = []
+
+        # dictionary for storing global (across the meshset list) of
+        # the metrics calculated
+        self._global_averages = {}
 
     def __get_entities(self, meshset):
-        if meshset is None or meshset == self.dagmc_file.root_set:
-            self.meshset_lst.append(self.dagmc_file.root_set)
-        else:
+        """convert the list of meshsets to its corresponding list of surfaces
+
+        inputs
+        ------
+            meshset: the list of meshsets on which query will be performed.
+
+        outputs
+        -------
+            none
+        """
+        # allow mixed list of surfaces and volumes that create a
+        # single list of all the surfaces together
+        if type(meshset) != list:
+            # convert single item to list for iterating
+            meshset = [meshset]
+        for m in meshset:
             dim = self.dagmc_file._my_moab_core.tag_get_data(
-                self.dagmc_file.dagmc_tags['geom_dim'], meshset)[0][0]
-            # get triangles of a volume
+                self.dagmc_file.dagmc_tags['geom_dim'], m)[0][0]
+            # get surfaces of a volume
             if dim == 3:
                 surfs = self.dagmc_file._my_moab_core.get_child_meshsets(
-                    meshset)
+                    m)
                 self.meshset_lst.extend(surfs)
-            # get triangles of a surface
+            # get surface
             elif dim == 2:
-                self.meshset_lst.append(meshset)
+                self.meshset_lst.append(m)
             else:
-                warnings.warn(
-                    'Meshset is not a volume nor a surface! Rootset will be used by default.')
-                self.meshset_lst.append(self.dagmc_file.root_set)
+                warnings.warn('Meshset is not a volume nor a surface!')
+
```

Comment:
Remove redundant surfaces (if user provides a volume and a surf that is part of that volume)
```
self.meshset_lst = list(set(self.meshset_lst))
```